src/api.js

```diff
@@ -182,6 +190,28 @@ export default class Api {
             changes: []
           };
         }
+
+        // Example response header:
+        // Next-Page:"https://syncto.dev.mozaws.net/v1/buckets/d83134db8e5cf927\
+        // 624742e63e5e9f8f/collections/history/records?_sort=last_modified&_li\
+        // mit=50&_token=1451832797410%3A50"
+        function getQueryParam(url, key) {
+          const queryStartPos = url.indexOf("?");
+          if (queryStartPos === -1) {
+            return;
+          }
+          const params = url.substring(queryStartPos + 1).split("#")[0].split("&");
+          for (let i = 0; i < params.length; i++) {
+            const pairs = params[i].split("=");
+            if (decodeURIComponent(pairs.shift()) == key) {
+              return decodeURIComponent(pairs.join("="));
+            }
+          }
+        }
+
+        const nextPageHeader = res.headers.get("Next-Page");
+        const fetchToken = (nextPageHeader ? getQueryParam(nextPageHeader, "_token") : null);
```

Comment:
I remember @Natim claiming we should not store the token but the whole url instead, though I can't remember the reasoning. Hint @Natim?